integration/integration_test.go

```diff
@@ -1101,6 +1107,130 @@ func TestPushImage(t *testing.T) {
 		require.NoError(t, err)
 	})
 
+	t.Run("CacheAndPushAuth", func(t *testing.T) {
```

Comment:
review: I may look into making this more of a table-driven test, as it's getting pretty unwieldy

Reply:
I was gonna suggest the same.

Reply:
I had a try and there's a bit too much difference behaviour between the various test cases to really make it worthwhile IMO compared with the cognitive overhead of DRYing this.